ddl/placement_rule_test.go

```diff
@@ -234,7 +234,19 @@ func (s *testPlacementSuite) TestPlacementBuild(c *C) {
 			c.Assert(err, IsNil)
 			got, err := json.Marshal(out.Rules)
 			c.Assert(err, IsNil)
-			c.Assert(out.Rules, DeepEquals, t.output, Commentf("%d test\nexpected %s\nbut got %s", i, expected, got))
+			for _, r1 := range t.output {
+				found := false
+				for _, r2 := range out.Rules {
+					if ok, _ := DeepEquals.Check([]interface{}{r1, r2}, nil); ok {
+						found = true
+						break
+					}
+				}
+				if !found {
+					c.Logf("%d test\nexpected %s\nbut got %s", i, expected, got)
+					c.Fail()
+				}
+			}
```

Comment:
It can only ensure `out.Rules` contains all rules in `t.output`, but cannot ensure that `t.output` contains all rules in `out.Rules`.
You need to check the length of the two outputs.

Reply:
Done.